packages/orchestrator-ui-components/src/components/WFOForms/UserInputForm.tsx

```diff
@@ -0,0 +1,602 @@
+    hasPrev = false,
+    userInput,
+}: IProps) {
+    const t = useTranslations('pydanticForms.userInputForm');
+    const { theme } = useOrchestratorTheme();
+    const { showConfirmDialog } = useContext(ConfirmationDialogContext);
+    const [processing, setProcessing] = useState<boolean>(false);
+    const [nrOfValidationErrors, setNrOfValidationErrors] = useState<number>(0);
+    const [rootErrors, setRootErrors] = useState<string[]>([]);
+
+    const openDialog = () => {
+        showConfirmDialog({
+            question: '',
+            confirmAction: () => {},
+            cancelAction: cancel,
+            leavePage: true,
+        });
+    };
+
+    const submit = async (userInput: any = {}) => {
+        if (!processing) {
+            setProcessing(true);
+
+            try {
+                await validSubmit(userInput);
+                setProcessing(false);
+                return null;
+            } catch (error: unknown) {
+                setProcessing(false);
+
+                if (
+                    axios.isAxiosError(error) &&
+                    error.response?.status === 400
+                ) {
+                    const json = error.response.data;
+                    setNrOfValidationErrors(json.validation_errors.length);
+                    setRootErrors(
+                        json.validation_errors
+                            .filter(
+                                (e: ValidationError) => e.loc[0] === '__root__',
+                            )
+                            .map((e: ValidationError) => e.msg),
+                    );
+                    throw Object.assign(new Error(), {
+                        details: json.validation_errors.map(
+                            (e: ValidationError) => ({
+                                message: e.msg,
+                                params: e.ctx || {},
+                                dataPath: '.' + e.loc.join('.'),
+                            }),
+                        ),
+                    });
+                }
+
+                // Let the error escape so it can be caught by our own onerror handler instead of being silenced by uniforms
+                setTimeout(() => {
+                    throw error;
+                }, 0);
+
+                // The form will clear the errors so also remove the warning
+                setNrOfValidationErrors(0);
+                setRootErrors([]);
+
+                // The error we got contains no validation errors so don't send it to uniforms
+                return null;
+            }
+        }
+    };
+
+    const onButtonClick = (
+        e:
+            | React.MouseEvent<HTMLButtonElement>
+            | React.MouseEvent<HTMLDivElement, MouseEvent>,
+        question: string | undefined,
+        confirm: ConfirmDialogActions['closeConfirmDialog'],
+    ) => {
+        console.log(e, question, confirm);
+        alert('TODO: Implement on buttonClick cancel with confirm modal');
```

Comment:
Fix this or create a ticket and add it to the TODO

Reply:
Created ticket https://github.com/workfloworchestrator/orchestrator-ui/issues/325 and added it to the comment